backend/amd64/reload.ml

```diff
@@ -169,6 +183,26 @@ method! reload_operation op arg res =
         done;
         arg'.(len - 1) <- r;
         (arg', [|r|])
+  | Iscalarcast (V128_to_scalar (Float64x2) | V128_of_scalar (Float64x2)) ->
+    (* These are just moves; either the argument or result may be on the stack. *)
+    begin match stackp arg.(0), stackp res.(0) with
+    | true, true -> ([| self#makereg arg.(0) |], res)
+    | _ -> (arg, res)
+    end
+  | Iscalarcast (V128_to_scalar (Float32x4) | V128_of_scalar (Float32x4)) ->
+    (* These do additional logic requiring the result to be a register.
+       CR mslater: (SIMD) replace once we have unboxed float32 *)
+    (arg, [| self#makereg res.(0) |])
+  | Iscalarcast (V128_of_scalar (Int64x2 | Int32x4 | Int16x8 | Int8x16)) ->
+    (* Int -> Vec regs need the result to be a register. *)
+    (arg, [| self#makereg res.(0) |])
+  | Iscalarcast (V128_to_scalar (Int64x2 | Int32x4)) ->
+    (* Vec -> Int regs need the argument to be a register. *)
+    ([| self#makereg arg.(0) |], res)
+  | Iscalarcast (V128_to_scalar (Int16x8 | Int8x16)) ->
+    (* These do additional logic requiring the result to be a register.
+       CR mslater: (SIMD) replace once we have unboxed int16/int8 *)
+    ([| self#makereg arg.(0) |], [| self#makereg res.(0) |])
```

Comment:
It is a bit unfortunate to have the
comments about the logics (only)
here since going forward we will
(at least) mainly use the CFG pipeline.

Reply:
This is true for old comments too, unrelated to simd. We should take care of all these comments in the next round of refactoring stack operands. There are many mentions of stack operands in https://github.com/ocaml-flambda/flambda-backend/issues/758 already, but none refer to comments explicitly, as far as I can tell, I'm going to add it there.